### PR TITLE
Add support for flatdata association mapping

### PR DESCRIPTION
**Pull request by PoojaParlapalli**
This is to enable association mapping for flat data(syntax similar to relational association mapping) .
with this multiple sections can be processed in a single service.

## How did you test this change?

- [x ] Test(s) added
- [ ] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

File: packages/legend-graph/src/graph/metamodel/pure/packageableElements/store/flatData/mapping/FlatDataAssociationPropertyMapping.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+export class FlatDataAssociationPropertyMapping
+  extends PropertyMapping
+  implements Hashable
+{
+  /**
+   * Studio does not process value specification, they are left in raw JSON form
+   *
+   * @discrepancy model
+   */
```
Comment (gayathrir11) [inline packages/legend-graph/src/graph/metamodel/pure/packageableElements/store/flatData/mapping/FlatDataAssociationPropertyMapping.ts:32]: Can you please remove above comments. It is not needed.

File: packages/legend-graph/src/graphManager/protocol/pure/v1/model/packageableElements/store/flatData/mapping/V1_FlatDataAssociationPropertyMapping.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+export class V1_FlatDataAssociationPropertyMapping
+  extends V1_PropertyMapping
+  implements Hashable
+{
+  /**
+   * Studio does not process value specification, they are left in raw JSON form
+   *
+   * @discrepancy model
+   */
```
Comment (gayathrir11) [inline packages/legend-graph/src/graphManager/protocol/pure/v1/model/packageableElements/store/flatData/mapping/V1_FlatDataAssociationPropertyMapping.ts:32]: Can you please remove above comments. It is not needed.

Comment (PoojaParlapalli) [reply packages/legend-graph/src/graphManager/protocol/pure/v1/model/packageableElements/store/flatData/mapping/V1_FlatDataAssociationPropertyMapping.ts]: Thank you I did it

File: packages/legend-graph/src/graph/metamodel/pure/packageableElements/store/flatData/mapping/FlatDataAssociationPropertyMapping.ts (hunk 1 above)
Comment (PoojaParlapalli) [reply packages/legend-graph/src/graph/metamodel/pure/packageableElements/store/flatData/mapping/FlatDataAssociationPropertyMapping.ts:32]: Thank you I did it.

**Comment by gayathrir11** [comment]
Please add some unit tests. Take a look at `ImportResolutionRoundtrip.test.ts`

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]
please update the description of this PR with what you did. 
Also confirming the reason we didn't add legend engine roundtrip tests is because there is not support for flatdata in legend-engine?

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]
please update the description of this PR with what you did. 
Also confirming the reason we didn't add legend engine roundtrip tests is because there is not support for flatdata in legend-engine?

**Comment by PoojaParlapalli** [comment]
we were able to do manual testing with engine code

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**